src/rules/font_face.rs

```diff
@@ -133,57 +165,18 @@ impl<'i> ToCss for UrlSource<'i> {
       format.to_css(dest)?;
       dest.write_char(')')?;
     }
-    Ok(())
-  }
-}
-
-/// The `format()` function within the [src](https://drafts.csswg.org/css-fonts/#src-desc)
-/// property of an `@font-face` rule.
-#[derive(Debug, Clone, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub struct Format<'i> {
-  /// A font format name.
-  #[cfg_attr(feature = "serde", serde(borrow))]
-  pub format: FontFormat<'i>,
-  /// The `supports()` function.
-  // TODO: did this get renamed to `tech()`?
-  pub supports: Vec<FontTechnology>,
-}
-
-impl<'i> Parse<'i> for Format<'i> {
-  fn parse<'t>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i, ParserError<'i>>> {
-    let format = FontFormat::parse(input)?;
-    let mut supports = vec![];
-    if input.try_parse(|input| input.expect_ident_matching("supports")).is_ok() {
-      loop {
-        if let Ok(technology) = input.try_parse(FontTechnology::parse) {
-          supports.push(technology)
-        } else {
-          break;
-        }
-      }
-    }
-    Ok(Format { format, supports })
-  }
-}
 
-impl<'i> ToCss for Format<'i> {
-  fn to_css<W>(&self, dest: &mut Printer<W>) -> Result<(), PrinterError>
-  where
-    W: std::fmt::Write,
-  {
-    self.format.to_css(dest)?;
-    if !self.supports.is_empty() {
-      dest.write_str(" supports ")?;
-      let mut first = true;
-      for technology in &self.supports {
-        if first {
-          first = false;
-        } else {
-          dest.write_char(' ')?;
+    let tech_len = self.tech.len();
+    if tech_len > 0 {
+      dest.whitespace()?;
+      dest.write_str("tech(")?;
+      for i in 0..tech_len {
+        self.tech[i].to_css(dest)?;
+        if tech_len - 1 != i {
+          dest.write_char(',')?;
         }
-        technology.to_css(dest)?;
       }
```

Comment:
you can do `self.tech.to_css(dest)?;` here - `Vec` implements the `ToCss` trait if its contents does.